packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -649,21 +649,37 @@ export class DataStores implements IDisposable {
      * This is called to update objects whose routes are unused. The unused objects are deleted.
      * @param unusedRoutes - The routes that are unused in all data stores in this Container.
      */
-    public updateUnusedRoutes(unusedRoutes: string[]) {
+    public updateUnusedRoutes(unusedRoutes: string[]): string[] {
+        const deletedRoutes = new Set<string>();
+
         for (const route of unusedRoutes) {
             const pathParts = route.split("/");
-            // Delete data store only if its route (/datastoreId) is in unusedRoutes. We don't want to delete a data
-            // store based on its DDS being unused.
+            const dataStoreId = pathParts[1];
+
+            // Skip any routes already deleted
+            if (deletedRoutes.has(route)) {
```

Comment:
Does this ever happen that there are duplicate routes here?

Reply:
I don't know, I don't think so based on what we do. Should I assert here?

Reply:
I don't think we should assert since it's not a big deal if that happens. What we can do is make sure that in GarbageCollector we remove duplicate routes. I think the `runGarbageCollection` function takes care of that but I am not sure.

Reply:
I'm pretty sure it's a non-issue, but I had this piece of code here just in case. We can log an event if we care at all.

Reply:
If we do somehow happen to delete the same route again, we will likely run into issues

Reply:
I'll turn this into a comment, and we can cover it in the validation PR